applications/ConvectionDiffusionApplication/python_scripts/convection_diffusion_stationary_embedded_solver.py

```diff
@@ -38,26 +44,77 @@ def AddVariables(self):
+            settings.AddEmptyValue("model_part_name").SetString(self.main_model_part.Name)
+            settings.AddEmptyValue("mls_extension_operator_order").SetInt(self.settings["mls_extension_operator_order"].GetInt())
+            settings.AddEmptyValue("avoid_zero_distances").SetBool(True)
+            settings.AddEmptyValue("deactivate_negative_elements").SetBool(True)
+            settings.AddEmptyValue("deactivate_intersected_elements").SetBool(False)
+            constraint_process = ConvectionDiffusionApplication.EmbeddedMLSConstraintProcess(self.model, settings)
+            constraint_process.Execute()
+
+        elif self.settings["use_distance_modification"].GetBool():
+            # FUTURE: use DistanceModificationProcess with tol_d
+            # Avoid small cuts
+            tol_d = 0.001
+            for node in self.GetComputingModelPart().Nodes:
+                d = node.GetSolutionStepValue(KratosMultiphysics.DISTANCE)
+                if abs(d) < tol_d:
+                    if d > 0.0:
+                        node.SetSolutionStepValue(KratosMultiphysics.DISTANCE, 0, tol_d)
+                    else:
+                        node.SetSolutionStepValue(KratosMultiphysics.DISTANCE, 0, -tol_d)
+
+            # Deactivate elements in negative distance region
+            for element in self.GetComputingModelPart().Elements:
+                n_pos = 0
+                for node in element.GetGeometry():
+                    if node.GetSolutionStepValue(KratosMultiphysics.DISTANCE) > 0.0:
+                        n_pos += 1
+                if n_pos == 0:
+                    element.Set(KratosMultiphysics.ACTIVE, False)
+                    for node in element.GetGeometry():
+                        node.Set(KratosMultiphysics.ACTIVE, False)
+
+        else:
+            # FUTURE: use DistanceModificationProcess with tol_d
+            # Avoid zero distances
+            tol_d = 1.0e-12
+            for node in self.GetComputingModelPart().Nodes:
+                d = node.GetSolutionStepValue(KratosMultiphysics.DISTANCE)
+                if abs(d) < tol_d:
+                    if d > 0.0:
+                        node.SetSolutionStepValue(KratosMultiphysics.DISTANCE, 0, tol_d)
+                    else:
+                        node.SetSolutionStepValue(KratosMultiphysics.DISTANCE, 0, -tol_d)
+
+            # Deactivate elements in negative distance region
+            for element in self.GetComputingModelPart().Elements:
+                n_pos = 0
                 for node in element.GetGeometry():
-                    node.Set(KratosMultiphysics.ACTIVE, False)
+                    if node.GetSolutionStepValue(KratosMultiphysics.DISTANCE) > 0.0:
+                        n_pos += 1
+                if n_pos == 0:
+                    element.Set(KratosMultiphysics.ACTIVE, False)
+                    for node in element.GetGeometry():
+                        node.Set(KratosMultiphysics.ACTIVE, False)
```

Comment:
Regardless the tolerance, this seems copy&paste to me. Which is the difference?

Reply:
It is copy&paste..
the idea was to call the DistanceModificationProcess once it is moved to the core.